spec/reek/configuration/app_configuration_spec.rb

```diff
@@ -7,48 +7,65 @@
 
 RSpec.describe Reek::Configuration::AppConfiguration do
   describe 'factory methods' do
-    let(:expected_excluded_paths) do
-      [SAMPLES_DIR.join('two_smelly_files'),
-       SAMPLES_DIR.join('source_with_non_ruby_files')]
-    end
-
-    let(:expected_default_directive) do
-      { Reek::SmellDetectors::IrresponsibleModule => { 'enabled' => false } }
+    around do |example|
```

Comment:
I think that in hindsight it was a huge mistake that we structured our specs so that they rely on external sample directories and external configuration directories since that makes it very hard to reason about them. Not to mention that a single spec is not self-reliant but has a whole dependency tree of files and directories it expects. I'm trying to reverse this development and I started with this file in the scope of this feature. As you can see, this spec is fully self-reliant and there are no external dependencies.

Reply:
Yes, I think it's a good idea to make this change. Perhaps we should add a note to the `CONTRIBUTING.md` file?

Reply:
Good idea, added a paragraph.